### PR TITLE
Fix sticky error state after invalid location search

### PR DESCRIPTION
**Pull request by brangi**
## Description
Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/10577

## Screenshots
Screen Shot 2020-08-19 at 3 42 01 PM

## Testing done

Locally, http://917c43b0c5ac86b06420538fc781b5f1.review.vetsgov-internal/ and e2e cypress test

## Acceptance criteria
- [x] The user should be able to continue searching after an error state without requiring a page refresh.

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/tests/e2e/search.cypress.spec.js [hunk 1]
```diff
@@ -88,4 +88,40 @@ describe('Facility search', () => {
...
+    cy.get('#street-city-state-zip').type('Austin, TX');
+    cy.get('#facility-type-dropdown').select('VA health');
+    cy.get('#service-type-dropdown').select('Primary care');
+    cy.get('#facility-search').click();
+    cy.get('#facility-search-results').contains(
+      'Results for "VA health", "Primary care" near "Austin, Texas"',
+    );
+    cy.get('.facility-result a').should('exist');
+
+    cy.injectAxe();
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/e2e/search.cypress.spec.js:124]: I'm pretty sure we only need to call `injectAxe` once per test.

File: src/applications/facility-locator/components/ResultsList.jsx [hunk 2]
```diff
@@ -115,35 +116,34 @@ class ResultsList extends Component {
...
+            <p>
+              If you need care right away for a minor illness or injury, select
+              Urgent care under facility type, then select either VA or
+              community providers as the service type.
+            </p>
+            <p>
+              If you have a medical emergency, please go to your nearest
+              emergency room or call 911.
+            </p>
+          </div>
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/ResultsList.jsx:142]: Instead of duplicating this block, can we move it into a separate component?

Comment (mpelzsherman) [reply src/applications/facility-locator/components/ResultsList.jsx:142]: OTOH we may want to consider displaying a different error for mapbox geocode errors, e.g.` "We couldn't find a matching location for <your input text here>."`

File: src/applications/facility-locator/components/ResultsList.jsx [hunk 3]
```diff
@@ -115,35 +116,34 @@ class ResultsList extends Component {
...
+        );
       }
-
+    } else if (currentQuery.error) {
+      // Mapbox query error
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/ResultsList.jsx:146]: How do we know this is a Mapbox error?

Comment (brangi) [reply src/applications/facility-locator/components/ResultsList.jsx:146]: I am pretty sure it is being sent by the catch

Comment (brangi) [reply src/applications/facility-locator/components/ResultsList.jsx:146]: I'll see what I can do making this error more descriptive

File: src/applications/facility-locator/components/ResultsList.jsx (hunk 2 above)
Comment (brangi) [reply src/applications/facility-locator/components/ResultsList.jsx:142]: good idea

**Comment by mpelzsherman** [comment]
There's a related scenario which should probably be handled in a separate PR:

If I search in zip code 27606 for, say `CCP / Poetry Therapist`, I get the error message “We’re sorry. We couldn’t complete your request." (This isn't technically true - it's just that there are no Poetry Therapists in my area, so we should consider changing that message.)

If I then switch the service type to “Clinic/Center - Primary Care”, the search seems to hang forever. As far as I can tell, it isn’t re-doing the search. I don’t see another api request in the network tab.

**Comment by brangi** [comment]
> There's a related scenario which should probably be handled in a separate PR:
> 
> If I search in zip code 27606 for, say `CCP / Poetry Therapist`, I get the error message “We’re sorry. We couldn’t complete your request." (This isn't technically true - it's just that there are no Poetry Therapists in my area, so we should consider changing that message.)
> 
> If I then switch the service type to “Clinic/Center - Primary Care”, the search seems to hang forever. As far as I can tell, it isn’t re-doing the search. I don’t see another api request in the network tab.

This is another bug, but I will no address it on this PR just what was reported by tzei.

File: src/applications/facility-locator/components/SearchResultMessage.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+function SearchResultMessage({
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/SearchResultMessage.jsx:3]: I know we are not consistent, but let's use es6 syntax for new code:

```suggestion
const SearchResultMessage = ({
 message,
  error,
  resultsFound,
  resultRef,
  facilityType
  }) => {
```

File: src/applications/facility-locator/components/SearchResultMessage.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        Please enter a location (street, city, state or postal code) and click
+        search above to find facilities.
+      </div>
+    );
+  };
+
+  return renderMessage(error, message, resultsFound, resultRef, facilityType);
+}
+
+export default SearchResultMessage;
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/SearchResultMessage.jsx:57]: Love the refactor! Can you please add PropTypes?

File: src/applications/facility-locator/components/SearchResultMessage.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    return (
+      <div className="search-result-title">
+        Please enter a location (street, city, state or postal code) and click
+        search above to find facilities.
+      </div>
+    );
+  };
+
+  return renderMessage(error, message, resultsFound, resultRef, facilityType);
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/SearchResultMessage.jsx:54]: I'm not sure we need the inner `renderMessage` function.

**Review by mpelzsherman — APPROVED** [review]

**Pull request merged**